src/Momento.Sdk.Incubating/Internal/ScsDataClient.cs

```diff
@@ -3,53 +3,278 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Google.Protobuf;
+using Microsoft.Extensions.Logging;
 using Momento.Protos.CacheClient;
+using Momento.Sdk.Config;
 using Momento.Sdk.Exceptions;
 using Momento.Sdk.Incubating.Responses;
 using Momento.Sdk.Internal;
 using Momento.Sdk.Internal.ExtensionMethods;
+using Momento.Sdk.Responses;
+using Momento.Sdk.Internal.Middleware;
+using Grpc.Core;
 
 namespace Momento.Sdk.Incubating.Internal;
 
 internal sealed class ScsDataClient : ScsDataClientBase
 {
-    public ScsDataClient(string authToken, string endpoint, uint defaultTtlSeconds, uint? dataClientOperationTimeoutMilliseconds = null)
-        : base(authToken, endpoint, defaultTtlSeconds, dataClientOperationTimeoutMilliseconds)
+    public ScsDataClient(IConfiguration config, string authToken, string endpoint, uint defaultTtlSeconds, ILoggerFactory loggerFactory)
+        : base(config, authToken, endpoint, defaultTtlSeconds, loggerFactory)
     {
     }
 
+    public async Task<CacheGetBatchResponse> GetBatchAsync(ISimpleCacheClient simpleCacheClient, string cacheName, IEnumerable<string> keys)
+    {
+        return await SendGetBatchAsync(simpleCacheClient, cacheName, keys.Select(key => key));
+    }
+
+    public async Task<CacheGetBatchResponse> GetBatchAsync(ISimpleCacheClient simpleCacheClient, string cacheName, IEnumerable<byte[]> keys)
+    {
+        return await SendGetBatchAsync(simpleCacheClient, cacheName, keys.Select(key => key));
+    }
+
+    public async Task<CacheGetBatchResponse> SendGetBatchAsync(ISimpleCacheClient simpleCacheClient, string cacheName, IEnumerable<string> keys)
+    {
+        // Gather the tasks
+        var tasks = keys.Select(key => simpleCacheClient.GetAsync(cacheName, key));
```

Comment:
I recommend pushing this back up to the to the `GetBatchAsync` functions above. That is, each `GetBatchAsyc` gathers the tasks, then makes a call to the same `SendGetBatchAsync` method. That way `SendGetBatchAsync` would only need to process the tasks, and we would only need one of those methods (not two)